geoindicators/src/test/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/TypologyClassificationTests.groovy

```diff
@@ -189,43 +192,77 @@ class TypologyClassificationTests {
+        def indicatorsPath = getClass().getResource("lczTests/trainingDatabdtopov2.geojson").toURI()
+        def indicatorsTable = "trainingDatabdtopov2"
+        h2GIS.load(indicatorsPath, indicatorsTable)
+
+        // Replace the id_rsu (coming from a specific city) by the id (coming from the true values of LCZ)
+        def allColumns = h2GIS.getTable(indicatorsTable).getColumns()
+        allColumns.remove("ID_RSU")
+        allColumns.remove("ID")
+
+        h2GIS.execute """DROP TABLE IF EXISTS $indicatorsRightId;
+                                    CREATE TABLE $indicatorsRightId 
+                                            AS SELECT id AS id_rsu, ${allColumns.join(",")}
+                                            FROM $indicatorsTable;"""
+
+        def lczIndicNames = ["GEOM_AVG_HEIGHT_ROOF"             : "HEIGHT_OF_ROUGHNESS_ELEMENTS",
+                             "BUILDING_FRACTION_LCZ"            : "BUILDING_SURFACE_FRACTION",
+                             "ASPECT_RATIO"                     : "ASPECT_RATIO",
+                             "GROUND_SKY_VIEW_FACTOR"           : "SKY_VIEW_FACTOR",
+                             "PERVIOUS_FRACTION_LCZ"            : "PERVIOUS_SURFACE_FRACTION",
+                             "IMPERVIOUS_FRACTION_LCZ"          : "IMPERVIOUS_SURFACE_FRACTION",
+                             "EFFECTIVE_TERRAIN_ROUGHNESS_LENGTH": "TERRAIN_ROUGHNESS_LENGTH"]
+
+        // Get into a new table the ID, geometry column and the 7 indicators defined by Stewart and Oke (2012)
+        // for LCZ classification (rename the indicators with the real names)
+        def queryReplaceNames = ""
+        lczIndicNames.each { oldIndic, newIndic ->
+            queryReplaceNames += "ALTER TABLE $lczIndicTable ALTER COLUMN $oldIndic RENAME TO $newIndic;"
+        }
+        h2GIS.execute """DROP TABLE IF EXISTS $lczIndicTable;
+                                CREATE TABLE $lczIndicTable 
+                                        AS SELECT id_rsu, the_geom, ${lczIndicNames.keySet().join(",")} 
+                                        FROM $indicatorsRightId;
+                                $queryReplaceNames"""
+
+
+        // The classification algorithm is called
+        def classifyLCZ = Geoindicators.TypologyClassification.identifyLczType()
+        if(!classifyLCZ([rsuLczIndicators   : lczIndicTable,
+                         rsuAllIndicators   : indicatorsRightId,
+                         normalisationType  : "AVG",
+                         mapOfWeights       : mapOfWeights,
+                         prefixName         : "test",
+                         datasource         : h2GIS])){
+            println "Cannot compute the LCZ classification."
+            fail()
+        }
+        def rsuLcz = classifyLCZ.results.outputTableName
+
+        h2GIS.eachRow("SELECT a.expected, a.lcz, b.* FROM $trainingDataTable a LEFT JOIN $rsuLcz b ON a.id = b.id_rsu"){row ->
+            def lczExpected = row.lcz.split(",")
```

Comment:
@j3r3m1 I think here `a.lcz` and `row.lcz.split(",")` should be replaced by `a.lcz_type` and `row.lcz_type.split(",")`.
Once replace and if the line 262 is replaced by `lczExpected += row.bdtopov2`, the test fails because `row.lcz1` and `row.lcz2` are null, but the test reach it end.